pages/login/register.js

```diff
@@ -0,0 +1,324 @@
+        futureTerms: [], 
+        designCentreSafety: false,
+        whmis: false, 
+        machineShop: false,
+    });
+
+    const [hasError, setHasError] = useState({
+        fullName: false,
+        phoneNumber: false, 
+        email: false, 
+        program: false, 
+        studentId: false,  
+        termStatus: false, 
+        memberType: false, 
+        subteam: false,
+        designCentreSafety: false,
+        whmis: false, 
+        machineShop: false,
+    });
+
+    const setErrorMessages = (formErrors) => {
+        validateField(formValues, formErrors, 'fullName');
+        validateField(formValues, formErrors, 'phoneNumber');
+        validateField(formValues, formErrors, 'email');
+        validateField(formValues, formErrors, 'program');
+        validateField(formValues, formErrors, 'studentId');
+        validateField(formValues, formErrors, 'termStatus');
+        validateField(formValues, formErrors, 'memberType');
+        validateField(formValues, formErrors, 'subteam');
+        validateField(formValues, formErrors, 'designCentreSafety');
+        validateField(formValues, formErrors, 'whmis');
+        validateField(formValues, formErrors, 'machineShop');
+        
+        setHasError(formErrors);
+    }
+
+    const handleSubmit = (evt) => {
+        evt.preventDefault();
+        console.log(formValues)
+        const formErrors = {...hasError};
+
+        clearErrorMessages(formErrors);
+        setErrorMessages(formErrors);
+
+        const formHasErrors = Object.values(formErrors).some(err => err);
+        if (!formHasErrors) {
+            loginTransition.setVisible(false);
+
+            const {fullName, phoneNumber, email, program, studentId, termStatus, memberType, 
+                subteam, designCentreSafety, whmis, machineShop, previousTerms, futureTerms} = formValues;
+
+            const fullNameParts = fullName.split(/\s+/);
+            updateUser(dispatch, {
+                name: {
+                    first: fullNameParts[0].trim(),
+                    last: fullNameParts[fullNameParts.length - 1].trim()
+                },
+                phone: phoneNumber.trim(),
+                program: program.trim(),
+                personalEmail: email.trim(),
+                studentId: studentId.trim(),
+                activeSchoolTerms: [...previousTerms, ...futureTerms],
+                termStatus,
+                memberType,
+                subteams: [subteam], // NOTE: As of March 2022, members can only select one option for subteam. Before this, members can select multiple subteams. We will keep subteams as an array for now for backwards-compatability and to prevent conflicts with Database data.
+                designCentreSafety,
+                whmis,
+                machineShop,
+            }, user._id, router).then(res => {
+                router.push("/")
+            });
+        }        
+    }
+
+    const handleInputChange = (name, value) => {
```

Comment:
There seems to be a decent amount of repeated code between this file and `returning-members.js`, this function for example. Could we reuse some code between the two files somehow?

Reply:
Shared logic in `handleInputChange` has been moved into a `util` methods file. Please see changes.

`handleFieldChange` looks fine to me as there is nothing more to move out to a `util` file.

Same for `handleSubmit`.

I feel that `setErrorMessages` could be kept in their respective files for now, even if their implementations are same for now. I imagine in the future, it's possible the form can get more complicated, so there may not be that much shared logic in `setErrorMessages` any more in the future.

If you have further suggestions on how I can extract common logic between these similar functions, please let me know.